homeassistant/components/rfxtrx/binary_sensor.py

```diff
@@ -34,6 +39,39 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+SENSOR_STATUS_ON = [
+    "Panic",
+    "Motion",
+    "Motion Tamper",
+    "Light Detected",
+    "Alarm",
+    "Alarm Tamper",
+]
+
+SENSOR_STATUS_OFF = [
+    "End Panic",
+    "No Motion",
+    "No Motion Tamper",
+    "Dark Detected",
+    "Normal",
+    "Normal Tamper",
+]
+
+DEVICE_TYPE_DEVICE_CLASS = {
+    "X10 Security Motion Detector": DEVICE_CLASS_MOTION,
+    "KD101 Smoke Detector": DEVICE_CLASS_SMOKE,
+    "Visonic Powercode Motion Detector": DEVICE_CLASS_MOTION,
+    "Alecto SA30 Smoke Detector": DEVICE_CLASS_SMOKE,
+    "RM174RF Smoke Detector": DEVICE_CLASS_SMOKE,
+}
+
+DEVICE_TYPE_OFF_DELAY = {
+    "RM174RF Smoke Detector": timedelta(seconds=60),
+    "KD101 Smoke Detector": timedelta(seconds=60),
+    "Alecto SA30 Smoke Detector": timedelta(seconds=60),
+}
+
```

Comment:
Not sure if it's possible, but would it perhaps be nicer to not use the strings defined in pyRFXtrx but packettype/subtype as these may be less variable? For readability the strings could be in the comment. It would make it less prone to errors when someone decides to change the string in the library.

Reply:
Same with the SENSOR_STATUS_ON/SENSOR_STATUS_OFF right? These are essentially subsets of the STATUS in pyRFXtrx?

Reply:
Yes, but the rule is to not rely on raw internal behavior of a protocol in hass. That belong in library. Imho this is already a bit borderline.

I had it by packet type first, but we use string abstractions for sensor values, commands and everything.

Reply:
Well if you approach it like I guess in an ideal world the library would be able to give a device type?
Doing it like this will work as long as it's double checked with an upgrade to a new version of the underlying library that the strings haven't changed.

Reply:
Well it's a balance. I suppose library could have provided a list of protocol values that should be considered to have a certain feature set.
Library should however not concern itself with home assistant specifics like that, there exist something like device_class that just so happen to be a string that can contain the value "smoke".
The api of this library does expose a "string" based API, so i think it's fine.

Reply:
I don't have a very strong opinion here, so I guess it's fine.